### ISSUE TITLE
[Blazor, .NET5] When publishing webassembly app with trimming enabled an error IL1012 is thrown

### ISSUE THREAD

**Issue by plachor** [mono/linker#1648]
Hi, I have IL1012 on trimming stage of Blazor app when package `HtmlSanitizer` is linked here [related issue](https://github.com/mganss/HtmlSanitizer/issues/248) 

```
ILLink : error IL1012: IL Linker has encountered an unexpected error. Please report the issue at https://github.com/mono/linker/issues [BlazorApp2\BlazorApp2.csproj]
  Fatal error in IL Linker
  Unhandled exception. System.ArgumentNullException: Value cannot be null. (Parameter 'input')
     at System.Text.RegularExpressions.ThrowHelper.ThrowArgumentNullException(ExceptionArgument arg)
     at System.Text.RegularExpressions.Regex.Matches(String input)
     at Mono.Linker.Steps.MarkStep.MarkTypeWithDebuggerDisplayAttribute(TypeDefinition type, CustomAttribute attribute)
     at Mono.Linker.Steps.MarkStep.MarkTypeSpecialCustomAttributes(TypeDefinition type)
     at Mono.Linker.Steps.MarkStep.MarkType(TypeReference reference, DependencyInfo reason, IMemberDefinition sourceLocationMember)
     at Mono.Linker.Steps.MarkStep.MarkField(FieldDefinition field, DependencyInfo& reason)
     at Mono.Linker.Steps.MarkStep.MarkEntireTypeInternal(TypeDefinition type, Boolean includeBaseTypes, DependencyInfo& reason, IMemberDefinition sourceLocationMember, HashSet`1 typesAlreadyVisited)
     at Mono.Linker.Steps.MarkStep.MarkEntireTypeInternal(TypeDefinition type, Boolean includeBaseTypes, DependencyInfo& reason, IMemberDefinition sourceLocationMember, HashSet`1 typesAlreadyVisited)
     at Mono.Linker.Steps.MarkStep.MarkEntireAssembly(AssemblyDefinition assembly)
     at Mono.Linker.Steps.MarkStep.InitializeAssembly(AssemblyDefinition assembly)
     at Mono.Linker.Steps.MarkStep.Initialize()
     at Mono.Linker.Steps.MarkStep.Process(LinkContext context)
     at Mono.Linker.Pipeline.ProcessStep(LinkContext context, IStep step)
     at Mono.Linker.Pipeline.Process(LinkContext context)
     at Mono.Linker.Driver.Run(ILogger customLogger)
     at Mono.Linker.Driver.Main(String[] args)
```

**Comment by mateoatr** [comment]
Thanks for reporting the issue. The linker had trouble processing a `DebuggerDisplayAttribute` that is using `null` for the value arg of the attribute's ctor (HtmlSanitizer references AngleSharp.Css which does [this](https://github.com/AngleSharp/AngleSharp.Css/blob/af910d963797f1788457c9e033aa63b432efed1e/src/AngleSharp.Css/Dom/Internal/Rules/CssFontFaceRule.cs#L11)), should be fixed by #1649.

**Comment by maliming** [comment]
hi @mateoatr 
Is there a temporary solution? Thanks.

**Comment by jaklithn** [comment]
Thanks for fast and helpful feedback! I removed the dependency to HtmlSanitizer and now the linker works as expected.

Suggestion: Would it be possible for the linker to feedback in what assembly the failure occurs? That would be very helpful to pinpoint the problem and avoid it.

**Comment by marek-safar** [comment]
The fix should also be included in 5.0.2 and newer service releases.
